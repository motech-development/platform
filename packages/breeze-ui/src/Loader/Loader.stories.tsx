import { withA11y } from '@storybook/addon-a11y';
import { select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import BaseStyles from '../BaseStyles/BaseStyles';
import Loader from './Loader';

const stories = storiesOf('Loader', module);
const colour = {
  Default: 'default',
  Secondary: 'secondary',
};

stories.addDecorator(withA11y);
stories.addDecorator(withKnobs);

stories.add('Basic loader', () => (
  <>
    <BaseStyles />

    <Loader
      colour={select('Colour', colour, 'default') as 'default' | 'secondary'}
    />
  </>
));
