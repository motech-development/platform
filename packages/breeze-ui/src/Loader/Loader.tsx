import React, { FC, memo } from 'react';
import styled from 'styled-components';

interface ICircle {
  $colour: 'default' | 'secondary';
}

const Circle = styled.circle<ICircle>`
  ${({ $colour }) => `
    animation: dash 1.5s ease-in-out infinite;
    stroke: ${$colour === 'default' ? '#2e9dc8' : '#fff'};;
    stroke-linecap: round;

    @keyframes dash {
      0% {
        stroke-dasharray: 1, 150;
        stroke-dashoffset: 0;
      }
      50% {
        stroke-dasharray: 90, 150;
        stroke-dashoffset: -35;
      }
      100% {
        stroke-dasharray: 90, 150;
        stroke-dashoffset: -124;
      }
    }
  `}
`;

const BaseLoader = styled.svg`
  animation: rotate 2s linear infinite;
  height: 50px;
  left: 50%;
  margin: -25px 0 0 -25px;
  position: absolute;
  top: 50%;
  width: 50px;

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }
`;

export interface ILoaderProps {
  className?: string;
  colour?: 'default' | 'secondary';
}

const Loader: FC<ILoaderProps> = ({ colour = 'default', className = '' }) => (
  <BaseLoader className={className} viewBox="0 0 50 50">
    <Circle
      $colour={colour}
      cx="25"
      cy="25"
      r="20"
      fill="none"
      strokeWidth="4"
    />
  </BaseLoader>
);

export default memo(Loader);
