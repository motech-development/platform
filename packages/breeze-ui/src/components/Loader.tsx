import { FC } from 'react';
import TTheme from '../utils/theme';

export interface ILoaderProps {
  className?: string;
  colour?: TTheme;
}

// { colour = 'default', className = '' }
const Loader: FC<ILoaderProps> = () => <div />;

export default Loader;
