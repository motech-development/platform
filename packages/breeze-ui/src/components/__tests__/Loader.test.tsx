import { render } from '@testing-library/react';
import Loader from '../Loader';

describe('Loader', () => {
  it('should render component with default classes', () => {
    const { container } = render(<Loader />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component with custom css classes', () => {
    const { container } = render(<Loader className="w-10 h-10" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
