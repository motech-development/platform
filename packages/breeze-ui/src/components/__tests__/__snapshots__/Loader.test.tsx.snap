// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render component with custom css classes 1`] = `
<svg
  class="animate-spin w-10 h-10"
  fill="none"
  viewBox="0 0 24 24"
>
  <circle
    class="opacity-25"
    cx="12"
    cy="12"
    r="10"
    stroke="currentColor"
    stroke-width="4"
  />
  <path
    class="opacity-50"
    d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
    fill="currentColor"
  />
</svg>
`;

exports[`Loader should render component with default classes 1`] = `
<svg
  class="animate-spin w-12 h-12"
  fill="none"
  viewBox="0 0 24 24"
>
  <circle
    class="opacity-25"
    cx="12"
    cy="12"
    r="10"
    stroke="currentColor"
    stroke-width="4"
  />
  <path
    class="opacity-50"
    d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
    fill="currentColor"
  />
</svg>
`;
